### PR TITLE
chore(transaction): Untie scheduling from multi status

### PR DESCRIPTION
**Pull request by dranikpg**
Moves part out of #2376 

Idea: We decide whether we have to schedule not based on our multi status (atomic multi), but solely based on the fact if COORD_SCHED is set

Goal: Being able to use ScheduleSingleHop()/Schedule() for multi transactions, and thus later allow single hop multi transactions

### CODE REVIEW

File: src/server/transaction.cc [hunk 1]
```diff
@@ -775,27 +799,29 @@ OpStatus Transaction::ScheduleSingleHop(RunnableType cb) {
...
-  DCHECK(IsAtomicMulti() || (coordinator_state_ & COORD_SCHED) == 0);  // Multi schedule in advance.
-
   cb_ptr_ = &cb;
 
-  if (IsAtomicMulti()) {
-    multi_->concluding = true;
+  // We can be already scheduled if we're part of a multi transaction. Note: If a multi tx isn't
+  // scheduled, we assume it's not mimicking the interface, but actually preparing a single hop.
+  bool scheduled = (coordinator_state_ & COORD_SCHED) > 0;
+  if (!scheduled) {
```
Comment (romange) [inline src/server/transaction.cc:807]: lets reverse the condition.

File: src/server/transaction.cc [hunk 2]
```diff
@@ -775,27 +799,29 @@ OpStatus Transaction::ScheduleSingleHop(RunnableType cb) {
     return RunSquashedMultiCb(cb);
   }
 
-  DCHECK(!cb_ptr_);
```
Comment (romange) [inline src/server/transaction.cc:778]: why did you remove this DCHECK?

**Comment by romange** [comment]
Can you please expand in the PR description what is the potential ability you unlock with this PR?

**Comment by dranikpg** [comment]
Can you please expand in the PR description what is the potential ability you unlock with this PR?

> Idea: We decide whether we have to schedule not based on our multi status (atomic multi), but solely based on the fact if COORD_SCHED is set

Goal: Being able to use ScheduleSingleHop()/Schedule() as a starter for Multi-Transactions

**Review by romange — APPROVED** [review]
